openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -0,0 +1,198 @@
+"""
+API for managing user preferences.
+"""
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from django.utils.translation import ugettext as _
+
+from ..api.account import AccountUserNotFound, AccountNotAuthorized
+from ..models import UserPreference, PreferenceNotFound, PreferenceValidationError, PreferenceUpdateError
+from ..serializers import UserSerializer
+
+
+def _get_user(requesting_user, username, allow_staff=False):
+    """
+    Helper method to return the user for a given username.
+    """
+    try:
+        existing_user = User.objects.get(username=username)
+    except ObjectDoesNotExist:
```

Comment:
I still wonder about what errors should be thrown. I am leaning towards ProfileUserNotFound ProfileNotAuthorized instead of reusing the account errors.